docs/cookbook/entities/custom-model-accessible-for-channel-admin.rst

```diff
@@ -0,0 +1,289 @@
+
+.. code-block:: php
+
+    /**
+      * @var ChannelInterface
+      * @ORM\ManyToOne(targetEntity="Sylius\Plus\Entity\ChannelInterface")
+      * @ORM\JoinColumn(name="channel_id", referencedColumnName="id", nullable=true)
+      */
+     protected $channel;
+
+* Create a proper migration for these changes.
+
+* Next, create a form type for your entity:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Form;
+
+    use Sylius\Bundle\ChannelBundle\Form\Type\ChannelChoiceType;
+    use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+    use Sylius\Plus\ChannelAdmin\Application\Provider\AvailableChannelsForAdminProviderInterface;
+    use Symfony\Component\Form\Extension\Core\Type\TextType;
+    use Symfony\Component\Form\FormBuilderInterface;
+
+    final class SupplierType extends AbstractResourceType
+    {
+        /** @var AvailableChannelsForAdminProviderInterface */
+        private $availableChannelsForAdminProvider;
+
+        public function __construct(
+            AvailableChannelsForAdminProviderInterface $availableChannelsForAdminProvider,
+            string $dataClass,
+            array $validationGroups = []
+        ) {
+            parent::__construct($dataClass, $validationGroups);
+
+            $this->availableChannelsForAdminProvider = $availableChannelsForAdminProvider;
+        }
+
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('name', TextType::class, [
+                    'label' => 'Name'
+                ])
+                ->add('channel', ChannelChoiceType::class, [
+                    'choices' => $this->availableChannelsForAdminProvider->getChannels(),
+                    'label' => 'sylius.ui.channel',
+                ])
+            ;
+        }
+
+        public function getBlockPrefix(): string
+        {
+            return 'supplier';
+        }
+    }
+
+.. code-block:: yaml
+
+    App\Form\SupplierType:
+        arguments: ['App\Entity\Supplier', ['sylius'], '@Sylius\Plus\ChannelAdmin\Application\Provider\AvailableChannelsForAdminProviderInterface']
+        tags: ['form.type']
+
+The ``Sylius\Plus\ChannelAdmin\Application\Provider\AvailableChannelsForAdminProviderInterface`` service allows getting a list of proper channels for the currently logged-in admin.
+
+Remember to register ``App\Form\SupplierType`` for resource:
+
+.. code-block:: yaml
+
+    sylius_resource:
+        resources:
+            app.supplier:
+                driver: doctrine/orm
+                classes:
+                    model: App\Entity\Supplier
+       +            form: App\Form\SupplierType
+
+
+1. Add your entity to admin menu:
+----------------------------------
+
+Adding a new entity to the admin menu is described in the section ``How to customize Admin Menu`` of :doc:`this guide </customization/menu>`.
+
```

Comment:
Like @AdamKasp says, this whole part should not be a separate point, but should be included into the first point of this cookbook :)